### ISSUE TITLE
IgnoreUnmappedTargetProperties as an array of strings (may be feature request)

### ISSUE THREAD

**Issue by skaluva** [mapstruct/mapstruct#2597]
Hi,
I have a target pojo where i have around 30 fields but source has only 10fields. Remaining fields im ending up specifying like this.
```
        @Mapping(target = "accountCurrency", ignore = true)
	@Mapping(target = "accountId", ignore = true)
	@Mapping(target = "accountName", ignore = true)
	@Mapping(target = "affiliateId", ignore = true)
	@Mapping(target = "brandId", ignore = true)
	@Mapping(target = "channelId", ignore = true)
	@Mapping(target = "email", ignore = true)
	@Mapping(target = "extUserName", ignore = true)
	@Mapping(target = "feId", ignore = true)
	@Mapping(target = "ipAddress", ignore = true)
	@Mapping(target = "kycStatus", ignore = true)
	@Mapping(target = "langId", ignore = true)
	@Mapping(target = "playerState", ignore = true)
	@Mapping(target = "playerStatus", ignore = true)
	@Mapping(target = "productId", ignore = true)
	@Mapping(target = "screenName", ignore = true)
	@Mapping(target = "timeZone", ignore = true)
	@Mapping(target = "userId", ignore = true)
	@Mapping(target = "vipLevel", ignore = true)
        etc...
```
I dont want to define `unmappedTargetPolicy = ReportingPolicy.IGNORE` at interface level as i have other methods for which it should throw error if i miss target props.

Is there a way to specify ignorable target props in an array of strings instead of field by field, something like this:
`@IgnorableTargetProperties ({array of strings})`

`@BeanMapping` expects only `unmappedSourceProperties` as array of strings but not target properties.

**Comment by skaluva** [comment]
Is there a way to support this instead of specifying property by property?

**Comment by filiphr** [comment]
There is currently no way to specify properties in an array currently. And we are not big fans in doing that at the moment. I think that we have already discussed this with the team and have decided against adding something like it.

What you could do though is one of the following:

# Define `unmappedTargetPolicy` on the `@BeanMapping`

You can explicitly define `@BeanMapping(unmappedTargetPolicy = ReportingPolicy.IGNORE)` and only ignore for that particular method.

# Use `BeanMapping#ignoreByDefault`

With this you will have to explicitly define all the properties that you want to get mapped. In a way it is similar to the first one

**Comment by filiphr** [comment]
>Could you please tell me in which version this is available.

Not sure what you are asking for @skaluva. You can use meta annotations in your version. If you are referring to the `@IgnoredProperties` then this is your own one, that's what I wrote in my example.

If you are referring to `BeanMapping#unmappedTargetPolicy` then this is only available in 1.5 (currently on 1.5.0.Beta1 is released)
